bin/ddg2pca

```diff
@@ -0,0 +1,405 @@
+    - group_dict: Dictionary mapping indices to groups.
+    - desired_group: The group(s) to filter by.
+    
+    Returns:
+    - Filtered list.
+    """
+    # If the desired group is "all", return the original data.
+    if desired_group.lower() == "all":
+        return data
+    
+    # If multiple groups are provided, split them.
+    desired_groups = [group.strip().capitalize() for group in desired_group.split(",")]
+    
+    # Get the indices that belong to the desired group.
+    valid_indices = [index for index, group in group_dict.items() if any(desired in group for desired in desired_groups)]
+    
+    # Filter the data based on the valid indices, but also ensure the index is within the bounds of the data list.
+    return [data[i] for i in valid_indices if i < len(data)]
+
+def set_default_config():
+    '''
+    If no config file is provided with custom grouping of residues, this will be the default plotting configuration
+    '''
+    config = {'Aromatic': {'residues': ['F', 'W', 'Y'], 'color': (0.00, 0.45, 0.70)},
+    'Acidic': {'residues': ['D', 'E'], 'color': (0.00, 0.62, 0.45)},
+    'Basic': {'residues': ['H', 'K', 'R'], 'color': (0.94, 0.89, 0.26)},
+    'Polar': {'residues': ['C', 'N', 'Q', 'S', 'T'], 'color': (0.90, 0.62, 0.00)},
+    'Hydrophobic': {'residues': ['A', 'G', 'I', 'L', 'M', 'P', 'V'],
+    'color': (0.34, 0.71, 0.91)}}
+
+    return config
+
+def read_textlist_file(text_file_path):
+    '''
+    Reads the textlist file with the residues whose labels should be seen in the plot
+    ''' 
+    with open(text_file_path, 'r') as f:
+        text_list = f.read().splitlines()
+    return text_list
+
+def get_ddg_data(options):
+    try:
+        res_order = parse_mutlist_file(options.mutation_list)      
+    except (IOError, TypeError):
+        exit(1)
+
+    log.info("Mutation list is:")
+    for i,r in enumerate(res_order):
+        log.info("%d\t%s" % (i,r))
+
+    try:
+        res_ids_str = get_residue_list(options.in_pdb, multimers=options.multimers)
+    
+    except IOError:
+        exit(1)
+
+    ### Poslist parsing
+    if options.position_list is not None:
+        log.info("Positions will be read from file")
+        try:
+            poslist = parse_poslist_file(options.position_list, res_ids_str)
+
+        except (IOError, TypeError):
+            exit(1)
+
+        res_ids_str = filter_reslist(res_ids_str, poslist)
+   
+
+
+    ### Textlist parsing - same mechanism as poslist
+    if options.text_list:
+        log.info("Positions will be read from file")
+        try:
+            textlist = parse_poslist_file(options.text_list, res_ids_str)
+            
+        except (IOError, TypeError):
+            exit(1)
+
+        textlist = filter_reslist(res_ids_str, textlist)
+
+        # Determine if textlist items are single strings or tuples
+        is_single_chain = len(textlist[0]) == 1 # textlist items are single strings or tuples   
+
+    else:
+        textlist = []
```

Comment:
There is actually a reason for this. Later in the code, I apply some adjustments for the textlist for 1) Change the names in the textlist according to the new labels if these are provided and 2) Subset the textlist according to the specified residue gropung if provided. I noticed that if I created an empty textlist, the rest of the code could be maintained as is. However, if I set the textlist to None, the steps 1 and 2 returned errors due to texlist being nonexistent, and I would have to implement in each of those steps a condition ("if textlist..") before the code that adjusts the texlist to avoid bugs in case of a None texlist. I thought the empty list was a cleaner way to do it. Based on this, let me know if you'd still prefer the None version and I will make the changes

Reply:
I think the question here is more on what behaviour the user will expect; in this case, an empty file would result in the default without textfile, e.g. all the labels rather than in no labels which I think would be the expected outcome

Reply:
Hmmm I see your point. That interpretation of the code did not come to my mind. I will change it to None then :)